packages/pipeline-editor/src/PipelineSubmissionDialog.tsx

```diff
@@ -42,23 +42,31 @@ export class PipelineSubmissionDialog extends Widget
       ) as HTMLInputElement).value,
       runtime_config: (document.getElementById(
         'runtime_config'
-      ) as HTMLInputElement).value
+      ) as HTMLInputElement).value,
+      filetype: document.getElementById('pipeline_filetype') as HTMLInputElement
     };
   }
 
   getHtml(props: any): HTMLElement {
     const htmlContent = document.createElement('div');
     const br = '<br/>';
     let runtime_options = '';
+    let filetype_options = '';
     const runtimes = props['runtimes'];
+    const filetypes = ['tgz', 'tar.gz', 'zip', 'yaml', 'yml', 'py'];
```

Comment:
since the first value here will be the "default" you should have `tar.gz` first top match the backend